foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/handlers/SlowQueryReporter.java

```diff
@@ -0,0 +1,28 @@
+package com.flipkart.foxtrot.core.querystore.handlers;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.flipkart.foxtrot.core.querystore.ActionEvaluationResponse;
+import com.flipkart.foxtrot.core.querystore.ActionExecutionObserver;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ *
+ */
+@Slf4j
+public class SlowQueryReporter implements ActionExecutionObserver {
+    @Override
+    public void handleExecution(ActionEvaluationResponse response) {
+        if(null != response.getException() || response.isCached()) {
+            return;
+        }
+        if (response.getElapsedTime() > 1000) {
```

Comment:
Maybe this can be made configurable per action ?

Reply:
Not in this merge request

Reply:
Done